packages/portalnetwork/src/subprotocols/beaconChain/types.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  ByteListType,
+  ByteVectorType,
+  ContainerType,
+  ListCompositeType,
+  UintBigintType,
+  UnionType,
+  BitVectorType,
+} from '@chainsafe/ssz'
+
+// todo: export
+
+/* ----------------- Alliases ----------------- */
+const Bytes4 = new ByteVectorType(4)
+const Bytes32 = new ByteVectorType(32)
+const Bytes48 = new ByteVectorType(48)
+const Bytes96 = new ByteVectorType(96)
+/* ----------------- Types ----------------- */
+
+const Slot = new UintBigintType(64)
+const Epoch = new UintBigintType(64)
+const CommitteeIndex = new UintBigintType(64)
+const ValidatorIndex = new UintBigintType(64)
+const Gwei = new UintBigintType(64)
+const GeneralizedIndex = new UintBigintType(64)
+const Root = Bytes32
+const Hash32 = Bytes32
+const Version = Bytes4
+const BLSPubKey = Bytes48
+const BLSSignature = Bytes96
+const DomainType = Bytes4
+const Domain = Bytes32
```

Comment:
I am adding the Typescript types for each SSZ type alias but what would the correct Typescript types be for the SSZ types mentioned here on lines 26-32? These are `ByteVectorTypes`. @ScottyPoi

Reply:
My guess is `UintXArray`s but not entirely sure

Reply:
They're always Uint8Arrays.  So a `Bytes32` is just a Uint8Array in TS/JS since we can't set a fixed array length in the interface.

Reply:
There are internal checkers in `@chainsafe/ssz` that will throw an error if you try to pass a Uint8Array of the incorrect length for the type.
```
Bytes32.serialize( Uint8Array.from( [1, 2, 3] ) )
```
Would throw an error because it expects a Uint8Array of length 32 and got one of length 3